producer.go

```diff
@@ -320,11 +348,44 @@ func (p *Producer) msgToString(msg any) string {
 	return stringMsg
 }
 
-func (p *Producer) validateMsg(msg any) ([]byte, error) {
+func (p *Producer) sendMsgToDls(msg any, headers map[string][]string, err error) {
+	internStation := getInternalName(p.stationName)
+	if p.conn.configUpdatesSub.StationSchemaverseToDlsMap[internStation] {
+		msgToSend := p.msgToString(msg)
+		timeSent := time.Now()
+		id := GetDlsMsgId(internStation, p.Name, time.Now().String())
+		headersForDls := make(map[string]string)
+		for k, v := range headers {
+			concat := strings.Join(v, " ")
+			headersForDls[k] = concat
+		}
+		schemaFailMsg := &DlsMessage{
+			ID:          id,
+			StationName: internStation,
+			Producer: ProducerDetails{
+				Name:         p.Name,
+				ConnectionId: p.conn.ConnId,
+			},
+			Message: MessagePayloadDls{
+				TimeSent: timeSent,
+				Data:     msgToSend,
+				Headers:  headersForDls,
+			},
+			CreationDate: timeSent,
+		}
+		msgToPublish, _ := json.Marshal(schemaFailMsg)
+		_ = p.conn.brokerConn.Publish(GetDlsSubject("schema", internStation, id), msgToPublish)
+
+		if p.conn.configUpdatesSub.ClusterConfigurations["send_notification"] {
+			p.sendNotification("Schema validation has failed", "Station: "+p.stationName+"\nProducer: "+p.Name+"\nError: "+err.Error(), msgToSend, schemaVFailAlertType)
+		}
+	}
+}
+
+func (p *Producer) validateMsg(msg any, headers map[string][]string) ([]byte, error) {
 	sd, err := p.getSchemaDetails()
 	if err != nil {
-		msgToSend := p.msgToString(msg)
-		p.sendNotification("Schema validation has failed", "Station: "+p.stationName+"\nProducer: "+p.Name+"\nError: "+err.Error(), msgToSend, schemaVFailAlertType)
+		p.sendMsgToDls(msg, headers, err)
```

Comment:
in this case we shouldn't send it to the DLS since it is not validation issue it is an internal bug